src/main/java/com/google/netpcapanalysis/dao/BucketDaoImpl.java

```diff
@@ -0,0 +1,228 @@
+package com.google.netpcapanalysis.dao;
+
+import com.google.netpcapanalysis.models.Flagged;
+import com.google.netpcapanalysis.models.PCAPdata;
+import java.util.*; 
+import java.io.*;
+import java.lang.*;
+import java.util.Map.Entry;
+
+import io.pkts.PacketHandler;
+import io.pkts.Pcap;
+import io.pkts.buffer.Buffer;
+import io.pkts.packet.Packet;
+import io.pkts.packet.TCPPacket;
+import io.pkts.packet.UDPPacket;
+import io.pkts.packet.IPPacket;
+import io.pkts.protocol.Protocol;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.netpcapanalysis.interfaces.dao.PCAPDao;
+import com.google.netpcapanalysis.interfaces.dao.PCAPParserDao;
+import com.google.netpcapanalysis.interfaces.dao.BucketDao;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class BucketDaoImpl implements BucketDao {
+  private ArrayList<PCAPdata> allPCAP; 
+  private ArrayList<PCAPdata> sortedPCAP; 
+  private String myip = "";
+
+  // Map to store <String Class, HashMap<> of protocols and frequencies >
+  private LinkedHashMap<String, HashMap<String, Integer>> bucketData;
+  
+  private String classA = "Class A";
+  private String classB = "Class B";
+  private String classC = "Class C";
+  private String classDE = "Class D & E";
+
+  public BucketDaoImpl(ArrayList<PCAPdata> packets) {
+    allPCAP = packets; 
+    orderIPs();
+    loadBuckets();    
+  }
+
+  public LinkedHashMap<String, HashMap<String, Integer>> getBuckets() {
+    return bucketData;
+  }
+ 
+  /* Put packets into MYIP, OUTIP format and order by destination IP (ascending). */
+  private void orderIPs(){
```

Comment:
maybe we can all make a UtilityPCAPDao that can hold any functions that manipulate PCAP data such as find myip, sorting PCAP files, or getting all unique IP's.